src/engines/kotor/module.cpp

```diff
@@ -519,6 +521,17 @@ void Module::setReturnEnabled(bool enabled) {
 	_ingame->setReturnEnabled(enabled);
 }
 
+void Module::setGlobalNumber(const Common::UString &id, int value) {
+	_globalNumbers[id] = value;
+}
+
+int Module::getGlobalNumber(const Common::UString &id) const {
+	if (_globalNumbers.find(id) != _globalNumbers.end())
+		return _globalNumbers[id];
```

Comment:
That doesn't work that way, you're still using `operator[]`.

`_globalNumbers.find(id)` returns an iterator you can use to get the value (if it's not the end iterator). See for example https://github.com/xoreos/xoreos/blob/master/src/engines/kotor/area.cpp#L402 .